contracts/router/modules/bridge/SynapseBridgeModule.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {OnlyDelegateCall} from "../pool/OnlyDelegateCall.sol";
+import {IBridgeModule} from "../../interfaces/IBridgeModule.sol";
+import {ILocalBridgeConfig} from "../../interfaces/ILocalBridgeConfig.sol";
+import {ISynapseBridge} from "../../interfaces/ISynapseBridge.sol";
+import {Action, BridgeToken, DefaultParams, SwapQuery} from "../../libs/Structs.sol";
+import {UniversalTokenLib} from "../../libs/UniversalToken.sol";
+
+contract SynapseBridgeModule is OnlyDelegateCall, IBridgeModule {
+    using UniversalTokenLib for address;
+
+    error SynapseBridgeModule__EqualSwapIndexes(uint8 index);
+    error SynapseBridgeModule__UnsupportedDepositAction(Action action);
+    error SynapseBridgeModule__UnsupportedRedeemAction(Action action);
+    error SynapseBridgeModule__UnsupportedToken(address token);
+
+    /// These need to be immutable in order to be accessed via delegatecall
+    ILocalBridgeConfig public immutable localBridgeConfig;
+    ISynapseBridge public immutable synapseBridge;
+
+    constructor(address localBridgeConfig_, address synapseBridge_) {
+        localBridgeConfig = ILocalBridgeConfig(localBridgeConfig_);
+        synapseBridge = ISynapseBridge(synapseBridge_);
+    }
+
+    /// @inheritdoc IBridgeModule
+    function delegateBridge(
+        address to,
+        uint256 chainId,
+        address token,
+        uint256 amount,
+        SwapQuery memory destQuery
+    ) external payable {
+        assertDelegateCall();
+        (ILocalBridgeConfig.TokenType tokenType, address bridgeToken) = localBridgeConfig.config(token);
+        // Use config.bridgeToken as the token address for the bridging purposes
+        if (bridgeToken == address(0)) revert SynapseBridgeModule__UnsupportedToken(token);
+        // Approve the bridge to spend the token
+        bridgeToken.universalApproveInfinity({spender: address(synapseBridge), amountToSpend: amount});
+        if (destQuery.routerAdapter == address(0)) {
+            // If no Router Adapter is set, no action is required on destination chain, use `deposit()` or `redeem()`
+            (tokenType == ILocalBridgeConfig.TokenType.Redeem ? synapseBridge.redeem : synapseBridge.deposit)(
```

Comment:
Even though this looks cool in choosing function sig bw `redeem` and `deposit`, might not be easiest to read

Reply:
Resolved in 2883557e2ad03599c6c9757b3fd561283a62bed6